src/Common/src/TypeSystem/IL/Stubs/ArrayMethodILEmitter.cs

```diff
@@ -109,6 +109,17 @@ private void EmitILForAccessor()
 
                     ILLocalVariable thisEEType = _emitter.NewLocal(eetypePtrType);
 
+                    ILCodeLabel typeCheckPassedLabel = _emitter.NewCodeLabel();
+
+                    // Codegen will pass a null hidden argument if this is a `constrained.` call to the Address method.
+                    // As per ECMA-335 III.2.3, the prefix suppresses the type check.
+                    // if (hiddenArg.IsNull)
+                    //     goto TypeCheckPassed;
+                    codeStream.EmitLdArga(1);
+                    codeStream.Emit(ILOpcode.call,
+                        _emitter.NewToken(eetypePtrType.GetKnownMethod("get_IsNull", null)));
```

Comment:
You should not need the call to IsNull. You can just do brfalse directly on the value.